Update linbox to 1.7.1

cxzhong opened the pull request:
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->
Because the new version of linbox contains all patches. So we can just delete the patches and bump the new version.


### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

cxzhong committed:
Update linbox to 1.7.1

cxzhong committed:
Remove obsolete patches for uint128_t registration and various fixes

cxzhong committed:
Merge branch 'sagemath:develop' into update-linbox

dimpase reviewed (approved):
testing on linux goes well too.

cxzhong committed:
Apply linbox patch to fix ambiguous implicit conversions from Givaro::Integer to long

This patch fixes compilation issues with clang/libc++ by ensuring coefficients
are properly converted via the coefficient field, avoiding ambiguous implicit
conversions when setting NTL polynomial coefficients.

Related to: https://github.com/linbox-team/linbox/pull/329

cxzhong committed:
Merge branch 'develop' into update-linbox
